plugins/tuya/src/main.ts

```diff
@@ -0,0 +1,284 @@
+                name: camera.name,
+                nativeId,
+                info: {
+                    manufacturer: 'Tuya',
+                    model: camera.model,
+                    serialNumber: nativeId
+                },
+                type: TuyaDevice.isDoorbell(camera)
+                    ? ScryptedDeviceType.Doorbell
+                    : ScryptedDeviceType.Camera,
+                interfaces: [
+                    ScryptedInterface.VideoCamera,
+                ]
+            };
+
+            if (TuyaDevice.isDoorbell(camera)) {
+                device.interfaces.push(ScryptedInterface.BinarySensor);
+            }
+
+            if (TuyaDevice.hasStatusIndicator(camera)) {
+                device.interfaces.push(ScryptedInterface.OnOff);
+            }
+
+            if (TuyaDevice.hasMotionDetection(camera)) {
+                device.interfaces.push(ScryptedInterface.MotionSensor);
+            }
+
+            // Device Provider
+
+            if (TuyaDevice.hasLightSwitch(camera)) {
+                device.interfaces.push(ScryptedInterface.DeviceProvider);
+            }
+
+            devices.push(device);
+        }
+
+        await deviceManager.onDevicesChanged({
+            providerNativeId: this.nativeId,
+            devices
+        });
+
+        // Update devices with new state
+
+        for (const device of devices) {
+            this.getDevice(device.nativeId).then(device => device?.updateState());
+        }
+
+        // Handle any camera device that have a light switch
+
+        for (const camera of this.api.cameras) {
+            if (!TuyaDevice.hasLightSwitch(camera)) {
+                continue;
+            }
+            const nativeId = camera.id + '-light';
+            const device: Device = {
+                providerNativeId: camera.id,
+                name: camera.name + ' Light',
+                nativeId,
+                info: {
+                    manufacturer: 'Tuya',
+                    model: camera.model,
+                    serialNumber: camera.id,
+                },
+                interfaces: [
+                    ScryptedInterface.OnOff,
+                ],
+                type: ScryptedDeviceType.Light,
+            }
+
+            await deviceManager.onDevicesChanged({
+                providerNativeId: camera.id,
+                devices: [device]
+            });
+        }
+    }
+
+    async getDevice(nativeId: string): Promise<TuyaCamera> {
+        if (this.cameras.has(nativeId)) {
+            return this.cameras.get(nativeId);
+        }
+
+        const camera = this.api.cameras.find(camera => camera.id === nativeId);
+        if (camera) {
+            const ret = new TuyaCamera(this, nativeId, camera);
+            this.cameras.set(nativeId, ret);
+            return ret;
+        }
+
+        throw new Error('device not found?');
+    }
+}
+
+export default createInstanceableProviderPlugin("Tuya", nativeId => new TuyaController(nativeId));
```

Comment:
I'm not sure this is necessary unless you want to explicitly support multiple logins? you may have just copied this from unifi?

Reply:
In Tuya IoT Cloud, you can create multiple projects and each of those projects can contain multiple devices, however not sure if anyone would really do that

Reply:
k up to you, it's common in large unifi deployments.

Reply:
I'll keep it in the mean time